### DISCUSSION TITLE
Dealing with CS8602 after asserting non-null

### DISCUSSION THREAD

**Discussion by lonix1** [fluentassertions/fluentassertions#1813] [Q&A]
I'm sure many of you have seen this:

````cs
person.Should().NotBeNull();
person.Age.Should().BeGreaterThan(21);        // CS8602: Dereference of a possibly null reference
````

The simple solution is to add the `!` operator:
````cs
person!.Age.Should().BeGreaterThan(21);
````

But there must be a better way.

Is anyone aware of an analyser to automatically handle this, or has written an analyser for this, or knows of some clever way to tweak `.editorconfig` to allow this CS8602 scenario?

**Comment by jnyrup** [comment]
Have you seen the existing discussion about this topic in #1115?

My recommendation is still just to disable CS8602 for unit test projects.

**Reply by jnyrup** [reply]
If you don't have the final say in the code base you're working in, you could add the code proposed in https://github.com/fluentassertions/fluentassertions/issues/1115#issuecomment-925869230 to your code base.

**Comment by POnakS** [comment]
You could also use other libraries, like xUnit `.Assert` class, or Shouldly's `.ShouldNotBeNull()`, which handle these cases correctly.
